sql/core/src/test/scala/org/apache/spark/sql/expressions/ExpressionInfoSuite.scala

```diff
@@ -156,4 +158,74 @@ class ExpressionInfoSuite extends SparkFunSuite with SharedSparkSession {
       }
     }
   }
+
+  test("Check whether should extend NullIntolerant") {
+    // Only check expressions extended from these expressions
+    val parentExpressionNames = Seq(classOf[UnaryExpression], classOf[BinaryExpression],
+      classOf[TernaryExpression], classOf[QuaternaryExpression],
+      classOf[SeptenaryExpression]).map(_.getName)
```

Comment:
It seems some of string exprs are filtered out by the condition above, e.g., `String2TrimExpression ` https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L684 Yea, it might be difficult to list a complete list of candidate expressions for this check though, I think its better to list all of them as much as possible.

Reply:
This checker only check the expressions extended from these expressions(`UnaryExpression`, `BinaryExpression`, `TernaryExpression`, `QuaternaryExpression` and `SeptenaryExpression`) because these expressions are NullIntolerant by default.

Reply:
Ur, I got it. okay and please leave some comments about that, too.